scout/adapter/mongo/query.py

```diff
@@ -13,6 +14,54 @@
 
 
 class QueryHandler(object):
+    def build_case_query(self, case_id=None, status=None, older_than=None):
+        """Build case query based on case id, status and analysis date
+
+        Args:
+            case_id(str): id of a case
+            status(list): one or more case status
+            older_than(int): to select cases older than a number of months
+            case_ids(list): a list of case _ids
+
+        Returns:
+            case_query(dict): query dictionary
+        """
+        case_query = {}
+        if case_id:
+            case_query["_id"] = case_id
+        if status:
+            case_query["status"] = {"$in": list(status)}
+        if older_than:
+            older_than_date = datetime.now() - timedelta(weeks=older_than * 4)  # 4 weeks in a month
+            case_query["analysis_date"] = {"$lt": older_than_date}
+        return case_query
+
+    def delete_variants_query(self, case_id, variants_to_keep=[], min_rank_threshold=None):
+        """Build a query to delete variants from a case
+
+        Args:
+            case_id(str): id of a case
+            variants_to_keep(list): a list of variant ids
+            min_rank_threshold(int): remove variants with rank lower than this number
+
+        Return:
+            variant_query(dict): query dictionary
+        """
+        variants_query = {}
+        case_subquery = {"case_id": case_id}
+
+        # Create query to delete all variants that shouldn't be kept or with rank higher than min_rank_threshold
+        if variants_to_keep or min_rank_threshold:
+            variants_query["$and"] = [case_subquery]
+            if variants_to_keep:
+                variants_query["$and"].append({"_id": {"$nin": variants_to_keep}})
+            if min_rank_threshold:
```

Comment:
Do you know how this will work for cases without rank scores? I know there are some cancer cases with millions of variants without rank score. Could be worth testing...

Reply:
It's a good question! Tested on the demo cancer case with this command:
`scout --loglevel INFO --demo delete variants -u clark.kent@mail.com -c cancer_test -status inactive`

And got some variants removed. But test cancer case does have rank scores: 😆 

![image](https://user-images.githubusercontent.com/28093618/100456368-8a796e00-30c0-11eb-9c03-acad865379d2.png)

If you take for instance this case with no rank scores: https://scout.scilifelab.se/cust087/normalcondor, checking any variant in the database it has score=0
![image](https://user-images.githubusercontent.com/28093618/100456649-0b386a00-30c1-11eb-911f-f35c5a591397.png)
Then no variant will be removed, unless you specify rank_score=0 in the command params.

But I guess it's good be should be so. We should perhaps establish another system to remove variants from these cases and the'll have to be removed in another way